code/modules/mob/say.dm

```diff
@@ -76,11 +75,21 @@
 	if(speaking.flags & INNATE)
 		return TRUE
 
-	//Language check.
+	// Language check.
 	for(var/datum/language/L in src.languages)
 		if(speaking.name == L.name)
 			return TRUE
 
+	// Try to translate with augment if it is installed and not broken.
+	if(ishuman(src))
+		var/mob/living/carbon/human/human = src
+		var/obj/item/organ/internal/augment/translator/translator = human.internal_organs_by_name[BP_AUG_TRANSLATOR]
+
+		if(translator && !translator.is_broken())
+			for(var/language_name in translator.languages)
+				if(speaking.name == language_name)
+					return TRUE
```

Comment:
i'm not 100% sure, but i think you can simplify this bit by replacing the for loop and the if with this:
`if(speaking.name in translator.languages)`

Reply:
Yes, it turns out you can do that and it's much more pleasing to the eye